code/datums/components/deployable_item.dm

```diff
@@ -70,6 +73,12 @@
 		if(!ishuman(user) || HAS_TRAIT(item_to_deploy, TRAIT_NODROP))
 			return
 
+		var/area/area = get_area(location)
+		var/turf/open/placement_loc = location
+		if(restricted_deployment && (!placement_loc.allow_construction || area.area_flags & NO_CONSTRUCTION)) // long ass check to prevent things like deployable shields on alamo
```

Comment:
check restricted_deployment before typecasting the area and doing all the other checks, as they are not needed if this var isn't true.

Reply:
Alright, went ahead and did what I think ya wanted